src/OrbitGl/App.cpp

```diff
@@ -1328,9 +1328,12 @@ Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> OrbitApp::LoadCaptureFro
 
 orbit_base::Future<ErrorMessageOr<void>> OrbitApp::MoveCaptureFile(
     const std::filesystem::path& src, const std::filesystem::path& dest) {
+  std::optional<absl::Duration> capture_length =
+      capture_file_info_manager_.GetCaptureLengthByPath(src);
   return thread_pool_->Schedule([src, dest]() { return orbit_base::MoveFile(src, dest); })
-      .ThenIfSuccess(main_thread_executor_,
-                     [this, dest] { capture_file_info_manager_.AddOrTouchCaptureFile(dest); });
+      .ThenIfSuccess(main_thread_executor_, [this, dest, capture_length] {
+        capture_file_info_manager_.AddOrTouchCaptureFile(dest, capture_length);
```

Comment:
You can also put the line `std::optional<absl::Duration> capture_length = capture_file_info_manager_.GetCaptureLengthByPath(src);` into this lambda, then you don't have to copy capture_length into the lambda.. 

Or combined with my last comment, simply: 
```
.ThenIfSuccess(main_thread_executor_, [this, dest, capture_length] {
    capture_file_info_manager_.AddOrTouchCaptureFile(dest, GetCaptureTime());
});